web/pdf_page_view.js

```diff
@@ -468,19 +472,15 @@ class PDFPageView {
     if (!this.textLayer) {
       return;
     }
-    this.structTreeLayer ||= new StructTreeLayerBuilder();
 
-    const tree = await (!this.structTreeLayer.renderingDone
-      ? this.pdfPage.getStructTree()
-      : null);
-    const treeDom = this.structTreeLayer?.render(tree);
+    const treeDom = await this.structTreeLayer.render();
```

Comment:
This is unfortunately not consistent with the old code, and may (likely) cause bugs in some edge-case!
Note how previously we'd only create and insert the structTree in the DOM *once*, however you've removed that logic here and I'm not at all convinced that it's safe/correct unfortunately.